text/2978-stack_based_vec.md

```diff
@@ -0,0 +1,281 @@
+- Feature Name: `stack_based_vec`
+- Start Date: 2020-09-27
+- RFC PR: [rust-lang/rfcs#2990](https://github.com/rust-lang/rfcs/pull/2990)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+This RFC, which depends and takes advantage of the upcoming stabilization of constant generics (min_const_generics), tries to propose the creation of a new "growable" vector named `ArrayVec` that manages stack memory and can be seen as an alternative for the built-in structure that handles heap-allocated memory, aka `alloc::vec::Vec<T>`.
+
+# Motivation
+[motivation]: #motivation
+
+`core::collections::ArrayVec<T>` has several use-cases and should be conveniently added into the standard library due to its importance.
```

Comment:
One item that could be added here is that it can be used in const-eval:
```rust
const fn make_vec() -> ArrayVec<i32, 10> {
    let mut v = ArrayVec::new();
    v.push(1).ok().unwrap(); // Using ok() so that unwrap is const.
    v.push(2).ok().unwrap(); // Using ok() so that unwrap is const.
    v.push(3).ok().unwrap(); // Using ok() so that unwrap is const.
    v.push(4).ok().unwrap(); // Using ok() so that unwrap is const.
    v
}

const VEC: ArrayVec<i32, 10> = make_vec();

// We can use VEC as a normal constant.
```

Reply:
Use-cases should be placed under the `Guide-level explanation` section